### PR TITLE
Delete event

### PR DESCRIPTION
**Pull request by RodrigoACG**
## Type of Change
- [x] New Feature
- [ ] Debugging
- [ ] Refactor
- [ ] Documentation

## PR Issue link if applicable
Refer to JIRA board KAN-33

## Describe Changes Made
In this PrRwe made a delete action that allows the user to delete events.

## PR Checklist
- [x] Added Reviewer
- [ ] Followed TDD, And Test are Passing

### CODE REVIEW

File: api/views.py [hunk 1]
```diff
@@ -244,4 +244,22 @@ def current_session(request):
...
\ No newline at end of file
+# might need add @login_required to event create so only logged in users can create events 
+@csrf_exempt
+def delete_event(request, user_id, event_id):
+    if request.method == 'DELETE':
+        try:
+            user = get_object_or_404(User, id=user_id)
+            event = get_object_or_404(Event, id=event_id, owner=user)
+            event.delete()
+            return JsonResponse({
```
Comment (zach-bergman) [inline api/views.py:255]: Nice I like this response!

File: api/views.py [hunk 2]
```diff
@@ -244,4 +244,22 @@ def current_session(request):
...
+                'message': 'Event deleted successfully',
+                'data': {
+                    'user_id': user.id,
+                    'username': user.username,
+                    'deleted_event_id': event_id
+                }
+            }, status=200)
+        except Event.DoesNotExist:
```
Comment (zach-bergman) [inline api/views.py:263]: Great error handling

**Review by zach-bergman — APPROVED** [review]

File: api/urls.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
     path('api/users/<int:user_id>/events/<int:event_id>/leave', views.leave_event, name='leave_event'),
     path('api/users/<int:user_id>/events', views.get_user_events, name='user_events'),
     path('api/users/<int:user_id>/events/<int:event_id>', views.get_one_event, name='one_event'),
+    path('api/users/<int:user_id>/events/<int:event_id>/delete', views.delete_event, name='delete_event'),
     path('api/events/<int:event_id>/attendees', views.event_attendees, name='users_attending_event'),
```
Comment (GBowman1) [inline api/urls.py:12]: thank you for keeping it seperated, easy to get lost in the routes

File: concertmate_be/settings.py [hunk 4]
```diff
@@ -38,6 +38,7 @@
     'django.contrib.staticfiles',
     'django.contrib.sessions',
     'api',
+    'rest_framework',
 ]
```
Comment (GBowman1) [inline concertmate_be/settings.py:42]: i see you added the rest_framework to apps are you looking to switch to using the framework

**Comment by GBowman1** [comment]
Looks like CI is failing because you added rest_framework to apps but did not add it the the requirements file

**Comment by GBowman1** [comment]
if you add `djangorestframework` to the requirements file @RodrigoACG @clydeautin and add a commit to this branch the CI should pass

**Comment by GBowman1** [comment]
CI dependencies are squared away i would say its safe to merge the test thats failing is unrelated and because a line in the test is looking for the data nested under something which its not nested uner

**Review by GBowman1 — APPROVED** [review]

**Pull request merged**